legion/engine/physics/mesh_splitter_utils/intersecting_polygon_organizer.hpp

```diff
@@ -0,0 +1,660 @@
+                //currentSupportEdge.position = transform.worldToLocalMatrix.MultiplyPoint(worldStartIntersection + startToEndIntersection);
+            }
+            else
+            {
+                nextSupportEdge = std::make_shared<MeshHalfEdge>(baseEdge->nextEdge->position, baseEdge->nextEdge->uv, owner);
+                generatedEdges.push_back(nextSupportEdge);
+            }
+
+            supportEdge = nextSupportEdge;
+            //----------------------------------- Create Intersection -------------------------------------------------------------//
+            //Vector3 intersectionPosition;
+            int maxData = effectedBoundaryEdges.size() - 1;
+            int currentIndex = i + 1;
+            float interpolant = (float)currentIndex / maxData;
+
+            math::vec3 IntersectionEdgePosition =
+                initialIntersectionPosition + (startToEndPosition * interpolant);
+
+            math::vec3 localIntersectionEdgePosition =
+                math::inverse(transform) * math::vec4(IntersectionEdgePosition, 1);
+
+            math::vec2 edgeUV = initialUVIntersection + startToEndUV * interpolant;
+
+            meshHalfEdgePtr intersectionEdge = std::make_shared<MeshHalfEdge>
+                (localIntersectionEdgePosition, edgeUV, owner);
+            intersectionEdge->isBoundary = true;
+            generatedEdges.push_back(intersectionEdge);
+
+            CreateAllignedQuad(currentSupportEdge, nextSupportEdge, baseEdge,
+                intersectionEdge, generatedEdges, owner);
+
+            IntersectionEdgeInfo intersectionEdgeInfo(
+                intersectionEdge);
+
+            generatedIntersectionEdges.push_back(intersectionEdgeInfo);
+
+            
+
+        }
+
+        void CreateAllignedQuad(meshHalfEdgePtr currentSupport, meshHalfEdgePtr nextSupport, meshHalfEdgePtr baseEdge,
+            meshHalfEdgePtr intersectionEdge, std::vector<meshHalfEdgePtr>& generatedEdges, SplittablePolygonPtr owner)
+        {
+            //create new supporttriangle located at next support
+            meshHalfEdgePtr supportTriangle = std::make_shared<MeshHalfEdge>(nextSupport->position, nextSupport->uv, owner);
+
+            //currentSupport-intersection-supporttriangle
+            MeshHalfEdge::ConnectIntoTriangle(currentSupport, supportTriangle, intersectionEdge);
+
+            //create new nextsupporttriangle located at currentsupport
+            meshHalfEdgePtr  nextSupportTriangle = std::make_shared<MeshHalfEdge>(currentSupport->position, currentSupport->uv, owner);
+
+            //nextsupporttriangle-nextSupport-baseEdge
+            MeshHalfEdge::ConnectIntoTriangle(nextSupportTriangle, baseEdge, nextSupport);
+
+            supportTriangle->SetPairing(nextSupportTriangle);
+
+            generatedEdges.push_back(supportTriangle);
+            generatedEdges.push_back(nextSupportTriangle);
+        }
+
+    };
+
+
+
+
+
+}
```

Comment:
I did not read through this file in depth. The main thing is to comment most of the functions. I have assumed here that the code is tested and works.